backend/app/services/session_service.py

```diff
@@ -117,6 +141,14 @@ def create_new_session(
         new_session.status = SessionStatus.started
         new_session.started_at = datetime.now(UTC)
 
+        # Update user's daily session counter
+        today = new_session.started_at.date()
+        if user_profile.sessions_created_today_date != today:
```

Comment:
This check should be included int he above session limit check to ensure that if a new day is started, the session counter resets and a new session is allowed.